prow/gcsupload/options.go

```diff
@@ -97,6 +97,7 @@ func BindOptions(options *Options, fs *flag.FlagSet) {
 	fs.StringVar(&options.DefaultRepo, "default-repo", "", "optional default repo for GCS path encoding")
 
 	fs.StringVar(&options.Bucket, "gcs-bucket", "", "GCS bucket to upload into")
```

Comment:
I'd really rather we just have `--prefix=gs://foo/bar/stuff` and let the software figure out that foo is the bucket, bar/stuff is the object prefix (and drop the `--gcs-bucket` flag). 

This is exactly how https://github.com/kubernetes/test-infra/blob/master/testgrid/cmd/updater/main.go#L82 works, any reason we cannot just reuse that logic here?

Reply:
Seems reasonable -- do you think that should be exposed at the human-user flag level or also in job config YAML?

Reply:
Ideally both

Reply:
So marhsal into a string, which we then set into a gcs.Path?

Reply:
Yeah, that's easy, we just don't have internal and external config really so it is a little bit messier :)